components/02-form-elements/typeahead/README.md

```diff
@@ -0,0 +1,56 @@
+### Typeahead
+A component that allows the user to search and select an option from a long list of options, with the ability to navigate suggested results using the keyboard.
+
+Falls back to a standard select when JavaScript is unavailable.
+
+Allows alternative search values for bilingual usage (e.g. A user with language set to Welsh can still search _United Kingdom_ for _Y Deyrnas Unedig_ when looking for countries, or a user can type _Scotland_ to select _United Kingdom_).
+
+related: [GDS Prototype](https://govuk-location-picker-demo.herokuapp.com/country-of-birth)
+
+### Usage
+The component enhances a select dropdown. It will ignore any options where the value is empty (i.e. Please select an option).
+
+When the JavaScript loads it will hide the select and show a text input. It will also change the for attribute on the label to match the id of the text input. When the user selects a value it will update the value of the select and fire an onchange event on the select. If the user types an exact match for a value (case insensitive) it will automatically select this value.
+
+Alternatives can by added as a comma separated list. When a user selects an option via an alternative the matched alternative will be shown in brackets next to the suggestion, however the main text and value will still be used (i.e. User searches _Wales_ - Suggestion will be _United Kingdom (Wales)_ - Selection will change input value to _United Kingdom_ and select value to _GB_).
+
```

Comment:
I think we should note that this pattern may require the ability to link to a form allowing the user to search a data set or enter an address / value manually
```suggestion
**Note:** variants of this pattern should:
- include a link to a form allowing the user to add an address 'manually'
- to enter a value without selecting from the dataset
- to search a dataset and select a value from the returned results (no-js for larger datasets)
```

/cc @charlotte-hirst

Reply:
I've merged in my address picker branch which has a prototype for an address lookahead (under address rather than typeaheads) which should satisfy the user being able to enter the address manually.

I will create an example where the user does not necessarily have to choose from a predefined list next week.